### DISCUSSION TITLE
Parsed Optimization Objects not found?

### DISCUSSION THREAD

**Discussion by Ryan7291** [idaholab/moose#28170] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Moose not able to find some objects in the optimization library even though the make file has "ALL_MODULES" checked. Of the optimization module objects those labeled 'Parsed...' usually have issues. Ex....

```
*** ERROR ***
A 'ParsedVectorVectorRealReductionReporter' is not a registered object.

If you are trying to find this object in a dynamically loaded library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.

```
-----------OTHER ERROR CODE W/ SAME MAKEFILE--------------------------

```
*** ERROR ***
A 'ParsedVectorReporter' is not a registered object.

If you are trying to find this object in a dynamically loaded library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.
```

**Comment by GiudGiud** [comment]
Hello

These  objects are kind of new. how old is your version of moose?
It should say in the header of any simulation

Guillaume

**Reply by Ryan7291** [reply]
This is what shows up 

`MOOSE Version:           git commit 6152da8bd7 on 2023-11-12`

**Reply by GiudGiud** [reply, marked as answer]
too old. please update moose and recompile your app
